src/stream/src/executor/source/source_executor.rs

```diff
@@ -543,14 +543,20 @@ impl<S: StateStore> SourceExecutor<S> {
             let source_reader = source_desc.source.clone();
             let (column_ids, source_ctx) = self.prepare_source_stream_build(&source_desc);
             let source_ctx = Arc::new(source_ctx);
+            let source_ctx_clone = source_ctx.clone();
             let mut build_source_stream_fut = Box::pin(async move {
                 let backoff = get_infinite_backoff_strategy();
                 tokio_retry::Retry::spawn(backoff, || async {
+                    let source_ctx_inner = source_ctx_clone.clone();
+                    source_ctx_inner.metrics.init_stream_reader_retry_count.with_guarded_label_values(&[
+                        &source_ctx_inner.actor_id.to_string(),
+                        &source_ctx_inner.source_name,
+                    ]).inc();
```

Comment:
Could we simply move this into `source_reader.build_stream`? And perhaps rename to `stream_reader_build_count`.

Reply:
Actually, I feel the figure is a little hard to understand. Why did it go up and down 🤔

![image](https://github.com/user-attachments/assets/57378a75-4935-4df1-8787-61ecf34e628c)

Reply:
I am struggling with it too, suspect somewhere calls a rate.

Reply:
I believe the reason is RW is reporting the metric from multiple nodes and the count value from each node varies, leading to the fluctuation. 
But it is irrelevant to the alert part, we always alert when the value is non-zero.